test/L2/L2PriceFeedWithoutRounds.t.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.23;
+
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import { OwnableUpgradeable } from "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import { Test } from "forge-std/Test.sol";
+import { L2PriceFeedWithoutRoundsFactory } from "src/L2/L2PriceFeedWithoutRoundsFactory.sol";
+import { L2PriceFeedWithoutRounds } from "src/L2/L2PriceFeedWithoutRounds.sol";
+
+contract L2PriceFeedWithoutRoundsV2Mock is L2PriceFeedWithoutRounds {
+    string public testVersion;
+
+    function initializeV2(string memory _version) public reinitializer(2) {
+        testVersion = _version;
+    }
+
+    function onlyV2() public pure returns (string memory) {
+        return "Hello from V2";
+    }
+}
+
+contract L2PriceFeedWithoutRoundsTest is Test {
+    L2PriceFeedWithoutRounds public l2PriceFeed;
+    L2PriceFeedWithoutRounds public l2PriceFeedImplementation;
+
+    address public priceFeedAdapter = 0x1038999DCf0A302Cc8Eed72fAeCbf0eEBfC476b0;
+
+    function setUp() public {
+        // deploy L2PriceFeedWithoutRoundsFactory Implementation contract
+        L2PriceFeedWithoutRoundsFactory l2PriceFeedFactoryImplementation = new L2PriceFeedWithoutRoundsFactory();
+
+        // deploy L2PriceFeedWithoutRoundsFactory contract via Proxy and initialize it at the same time
+        L2PriceFeedWithoutRoundsFactory l2PriceFeedFactory;
+        l2PriceFeedFactory = L2PriceFeedWithoutRoundsFactory(
+            address(
+                new ERC1967Proxy(
+                    address(l2PriceFeedFactoryImplementation),
+                    abi.encodeWithSelector(l2PriceFeedFactory.initialize.selector)
+                )
+            )
+        );
+
+        // create L2PriceFeedWithoutRounds contract
+        l2PriceFeed =
+            L2PriceFeedWithoutRounds(l2PriceFeedFactory.createL2PriceFeedWithoutRounds("USDT", priceFeedAdapter));
+        assert(address(l2PriceFeed) != address(0));
+        assertEq(l2PriceFeed.decimals(), 8);
+        assertEq(keccak256(bytes(l2PriceFeed.description())), keccak256(bytes("Redstone Price Feed")));
+        assertEq(l2PriceFeed.getDataFeedId(), bytes32("USDT"));
+        assertEq(address(l2PriceFeed.getPriceFeedAdapter()), priceFeedAdapter);
+
+        // accept ownership
+        l2PriceFeed.acceptOwnership();
+
+        // check L2PriceFeedWithoutRoundsFactory variables
+        assertEq(l2PriceFeedFactory.l2PriceFeedWithoutRoundsContracts(0), address(l2PriceFeed));
+        assertEq(l2PriceFeedFactory.l2PriceFeedWithoutRoundsDataFeedIds(address(l2PriceFeed), 0), "USDT");
```

Comment:
These values can also be defined as constants but can be part of the refactor PR.

Reply:
Added suggestion to the [refactorization GH issue](https://github.com/LiskHQ/lisk-contracts/issues/209).